src/ExpressionEvaluator/Core/Source/ExpressionCompiler/InstructionDecoder.cs

```diff
@@ -50,11 +48,11 @@ internal string GetName(TMethodSymbol method, bool includeParameterTypes, bool i
             AppendFullName(builder, method);
 
             // parameter list...
-            var includeArgumentValues = argumentValues != null;
+            var parameters = method.Parameters;
+            var includeArgumentValues = argumentValues?.Count == parameters.Length;
```

Comment:
I'll expand out the condition, because I waffled myself on whether to use conditional member access here.